cmd/root_test.go

```diff
@@ -18,6 +21,49 @@ import (
 	"go.k6.io/k6/lib/testutils"
 )
 
+type blockingTransport struct {
+	fallback       http.RoundTripper
+	forbiddenHosts map[string]bool
+	counter        uint32
+}
+
+func (bt *blockingTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	host := req.URL.Hostname()
+	if bt.forbiddenHosts[host] {
+		atomic.AddUint32(&bt.counter, 1)
+		panic(fmt.Errorf("trying to make forbidden request to %s during test", host))
+	}
+	return bt.fallback.RoundTrip(req)
+}
+
+func TestMain(m *testing.M) {
+	exitCode := 1 // error out by default
+	defer func() {
+		os.Exit(exitCode)
+	}()
+
+	bt := &blockingTransport{
+		fallback: http.DefaultTransport,
+		forbiddenHosts: map[string]bool{
+			"ingest.k6.io":    true,
+			"cloudlogs.k6.io": true,
+			"app.k6.io":       true,
+			"reports.k6.io":   true,
+		},
+	}
+	http.DefaultTransport = bt
```

Comment:
Hmm I can add cleanup in a `defer`, but there is literally nothing after `TestMain`, the test exits. On the flip side, this is why we can also detect leaked goroutines in https://github.com/grafana/k6/pull/2803

Reply:
In general I agree, but IIRC, Go compiles and tests different packages in isolation, so it shouldn't matter. But even if it didn't, I'd argue it's probably better to have this `http.DefaultTransport` for as many tests as possible, just in case. To intentionally not revert it back.

There is no way normal k6 behavior will be affected, only tests, and it will either break a problematic test or do nothing, so no downsides. I even wondered if I shouldn't also `os.Setenv()` a real `HTTP_PROXY` environment variable and implement a small blocking proxy, just in case some part of `cmd/` uses a custom `http.RoundTripper` (that doesn't rely on `http.DefaultTransport`) in the future, but decided that might be a bit too far to go and not worth it just for a bit of extra assurance :sweat_smile: